### ISSUE TITLE
More Accesibility issues

### ISSUE THREAD

**Issue by ferrannp** [jquense/react-widgets#68]
Running a accessibility tool, I got several violations using Date and Time Picker. They are the following:

**Violation**: Each form control must have associated text.
Inputs like:
```<input class="rw-input" type="text" data-reactid=".0.3.3.0.1.1.0.$=13:0.0.1.0.1.0.0.2.1.2.0" aria-readonly="false" aria-disabled="true" value="" role="combobox" disabled="" placeholder="Start Date" aria-haspopup="true" aria-owns="myID_cal myID_time_listbox" aria-busy="false" aria-expanded="false">```

**Violation**: The WAI-ARIA property identifier is not supported global property on element identifier that does not have a WAI-ARIA role.
Buttons like:
```<button class="rw-btn" data-reactid=".0.3.3.0.1.1.0.$=13:0.0.1.0.1.0.0.2.1.2.3.0.1.$month_2.1.$week_0.$day_0.0" type="button" disabled="" aria-disabled="true" aria-selected="false" tabindex="-1">01</button>```

**Violation**: Elements having an aria-activedescendant property must also contain a tabindex attribute with a value greater than or equal to zero.
Tables like
``<table id="myID_cal_view" class="rw-calendar-grid" data-reactid=".0.3.3.0.1.1.0.$=13:0.0.1.0.1.0.0.2.1.2.3.0.1.$month_2" aria-activedescendant="myID_cal_view_selected_item" role="grid" aria-labelledby="myIDt_cal_view_label" disabled="">```

**Violation**: A WAI-ARIA widget must have an accessible name (see message help text for details).
ul like:
``<ul id="myID_time_listbox" class=" rw-list" data-reactid=".0.3.3.0.1.1.0.$=13:0.0.1.0.1.0.0.2.1.3.2.0.0" role="listbox" step="30" max="Thu Dec 31 2099 00:00:00 GMT+0100 (CET)" min="Mon Jan 01 1900 00:00:00 GMT+0100 (WMT)" style="max-height:200px;height:auto;" aria-hidden="true">``

**Violation**: Use the HTML5 attribute or the associated WAI-ARIA attribute (not both) on any one input element.
Elements like:
```<input class="rw-input" type="text" data-reactid=".0.3.3.0.1.1.0.$=13:0.0.1.0.1.0.0.2.1.2.0" aria-readonly="false" aria-disabled="true" value="" role="combobox" disabled="" placeholder="Start Date" aria-haspopup="true" aria-owns="myID_cal myID_time_listbox" aria-busy="false" aria-expanded="false">```

```<button class="rw-btn" data-reactid=".0.3.3.0.1.1.0.$=13:0.0.1.0.1.0.0.2.1.2.3.0.1.$month_2.1.$week_0.$day_4.0" type="button" disabled="" aria-disabled="true" aria-selected="false" tabindex="-1">05</button>```

**Potential violation**: Elements that use 'role' must contain required child elements for the role.
Inputs like:
```<input class="rw-input" type="text" data-reactid=".0.3.3.0.1.1.0.$=13:0.0.1.0.1.0.0.2.1.3.0" aria-readonly="false" aria-disabled="true" value="" role="combobox" disabled="" placeholder="End Date" aria-haspopup="true" aria-owns="myID_cal myID_time_listbox" aria-busy="false" aria-expanded="false">```

**Potential violation**: WAI-ARIA widget elements must have a onkeydown or onkeypress event handler in order to be keyboard operable.
Elements like:
```<ul id="myID_time_listbox" class=" rw-list" data-reactid=".0.3.3.0.1.1.0.$=13:0.0.1.0.1.0.0.2.1.2.2.0.0" role="listbox" step="30" max="Thu Dec 31 2099 00:00:00 GMT+0100 (CET)" min="Mon Jan 01 1900 00:00:00 GMT+0100 (WMT)" style="max-height:200px;height:auto;" aria-hidden="true">```

```<tr data-reactid=".0.3.3.0.1.1.0.$=13:0.0.1.0.1.0.0.2.1.3.3.0.1.$month_2.1.$week_1" role="row">```

**Potential violation**: Use header elements where appropriate.
```<div id="myID" class="start-end-calendar rw-datetimepicker rw-widget rw-state-disabled rw-has-both" data-reactid=".0.3.3.0.1.1.0.$=13:0.0.1.0.1.0.0.2.1.3" tabindex="-1" aria-label="End Date" aria-required="false">```

-----

With the great job you did with accessibility in this library and the keyboard navigation, I think it is worth to check those issues, they don't seem really complicated.

Thank you!

**Comment by jquense** [comment]
Hey there,

thanks for this. I will look into it! Some of these are definately false positives...the ones complaining about "associated text" can be fixed by providing a `name=` prop.

Thanks again, i will do more research and se if I can clean these up :)

**Comment by ferrannp** [comment]
Hey, thank you for so fast response. I tried a name as a prop, now I have an input like this:

Now looks like: 
```<input class="rw-input" type="text" data-reactid=".0.3.3.0.1.1.0.$=13:0.0.1.0.1.0.0.2.1.2.0" aria-readonly="false" aria-disabled="true" value="" role="combobox" name="someName" placeholder="Start Date" aria-haspopup="true" aria-owns="myID_cal myID_time_listbox" aria-busy="false" aria-expanded="false" disabled="true">```

But I am still getting same violations (this one is the first one).

**Comment by jquense** [comment]
Maybe i just don't understand what the error is then. as far as I can tell according the spec a name is an associated text...It might mean you also need a label associated with it

**Comment by ferrannp** [comment]
I think it talks about labels:

>  Each FORM control must have associated labels that are not of type "hidden". INPUT controls of type "submit", "reset", and "button" require text to be set in the "value" attribute. INPUT controls of type "image" require text to be set in the "alt" attribute. All other FORM controls are associated with text via the LABEL element. A LABEL is attached to a specific form control through the use of the "for" attribute. The value of the "for" attribute must be the same as the value of the "id" attribute of the form control. The aria-labelledby property can also be used to associate label for form controls.

**Comment by jquense** [comment]
Yeah ok, So the correct thing to do would be to pass the `id` attribute down to the widget input so you could actually create a label for it, since I can't change that without breaking existing code, I think the next best thing is to make sure the `labelled-by` attribute gets passed to the input. look for that in the next release hopefully.
